File: product-web-app/src/models/UserProduct.ts
import type { Product } from "./Product";

export interface UserProduct {
    id: number;
    userId: string;
    productId: number;
    product?: Product;
    purchasePrice: number;
    status: "pendiente" | "comprado" | "utilizado";
    createdAt: Date;
}

export interface CreateUserProductDto {
    productId: number;
    purchasePrice: number;
    status: "pendiente" | "comprado" | "utilizado";
}

File: ProductsAPI/DomainLayer/Models/UserProduct.cs
namespace DomainLayer.Models
{
    public enum ProductStatus
    {
        Pendiente,
        Comprado,
        Utilizado
    }

    public class UserProduct
    {
        public int Id { get; set; }
        public string UserId { get; set; }
        // public ApplicationUser User { get; set; }
        public int ProductId { get; set; }
        public string? Name { get; set; }
        public string? Description { get; set; }
        public decimal? PurchasePrice { get; set; }
        public ProductStatus Status { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}
